source/world/level/Level.hpp

```diff
@@ -184,10 +192,8 @@ class Level : public LevelSource
 	std::vector<LevelListener*> m_levelListeners;
 	ChunkSource* m_pChunkSource;
 	LevelStorage* m_pLevelStorage;
-	LevelData m_levelData;
```

Comment:
Frankly most of this stuff should be privated.  I wouldn't move the LevelData struct to make just it private, I'd place a private and a public label before and after it.